maze_dataset/tokenization/util.py

```diff
@@ -146,3 +151,125 @@ def coords_to_strings(
         else:
             result.extend(coord_to_strings_func(coord))
     return result
+
+
+def connection_list_to_adj_list(
+    conn_list: ConnectionList, shuffle_d0: bool = True, shuffle_d1: bool = True
+) -> Int8[np.ndarray, "conn start_end coord"]:
+    n_connections = conn_list.sum()
```

Comment:
extremely minor nitpicks:
- The type hint here could look like `Int8[np.ndarray, "conn start_end=2 coord"]:`
- `n_connections` should have an `int` type hint
- docstring would be good